flink-runtime/src/main/java/org/apache/flink/runtime/net/SSLUtils.java

```diff
@@ -163,80 +163,157 @@ public static void setSSLVerifyHostname(Configuration sslConfig, SSLParameters s
 	}
 
 	/**
-	 * Creates the SSL Context for the client if SSL is configured.
+	 * SSL engine provider.
+	 */
+	public enum SSLProvider {
+		JDK,
+		/**
+		 * OpenSSL with fallback to JDK if not available.
+		 */
+		OPENSSL;
+
+		public static SSLProvider fromString(String value) {
+			Preconditions.checkNotNull(value);
+			if (value.equalsIgnoreCase("OPENSSL")) {
```

Comment:
provide a constructor like `SSLProvider(String provider)` to give the enum's string representation looks better than hard code.